### ISSUE TITLE
sort-keys: add option to only apply sort to specifically named Objects

### ISSUE THREAD

**Issue by EmilyRosina** [eslint/eslint#14810]
**What rule do you want to change?**
- [sort-keys](https://eslint.org/docs/rules/sort-keys)

**Does this change cause the rule to produce more or fewer warnings?**
- More, _if opted into_.

**How will the change be implemented? (New option, new default behavior, etc.)?**
- New option.
- This added opinion will allow devs to define the Object name/key to perform the sort upon.
  This will also mean that the sort will not apply to any Objects with keys not matching the keys provided.
  I'm not sure if it can just be an Array with keys as Strings, or if it should be an Object, to then pass `error`, `off` etc.
  _Happy to get people's opinions on it, as to what's best_ :eyes: 

**Please provide some example code that this change will affect:**

> :information_source: For context, the code below is for a Vue component.

<details>
  <summary>Vue methods</summary>
  
> :information_source: I am unsure how `...` spread syntax will work with the sort.. but we need it to remain at the top, in our codebase.
> But I suppose as it has no actual key itself, then it would default to being at the top :thinking: :woman_shrugging: 
> 
> **Option example**
> ```js
>   "sort-keys": {
>     "objectNames": ["methods"]
>   }
> ```
```js
  methods: {
    ...mapActions('organization', [
      'GET_ORG_SUMMARY',
    ]),
    navigateTo (name) {
      const { orgCanonical } = this
      return { name, params: { orgCanonical } }
    },
    getOrganization (canonical) {
      this.GET_ORG_SUMMARY({ orgCanonical: canonical })
    },
  },
```
</details>

<details>
  <summary>Vue i18n</summary>
  
> :information_source: I wonder also if we should be able to define keyPaths, or it's just assumed to apply to nested Object items...
> As here we'd actually want to sort both `i18n.messages` and all of the keys nested under `en`, `es` and `fr`.
>
> **Option example**
> ```js
>   "sort-keys": {
>     "objectNames": ["i18n"]
>   }
> ```
```js
  i18n: {
    messages: {
      en: {
        title: 'Dashboard',
        catalogRepositories: 'Catalog Repositories',
        createOrgBtn: 'Create an organization',
        credentials: 'Credentials',
        noOrgFound: `You don't belong to an organisation`,
      },
      es: {
        title: 'Panel de control',
        catalogRepositories: 'Repositorio de los catálogos',
        createOrgBtn: 'Crear una organización',
        credentials: 'Credentials',
        noOrgFound: 'Usted no pertenece a ninguna organización',
      },
      fr: {
        title: 'Tableau de bord',
        catalogRepositories: 'Sources du catalogue',
        createOrgBtn: 'Créer une organisation',
        credentials: `Informations d'identification`,
        noOrgFound: `Vous n'appartenez à aucune organisation`,
      },
    },
  },
```
</details>

**What does the rule currently do for this code?**
- It applies to ALL Objects, with no way to filter.
  So we cannot use this rule at the moment, as some Objects we have other sort orders 
  we wish to follow instead of sorting alphabetically.

**What will the rule do after it's changed?**
- Add the option to provide specifically named Objects on which the sort will apply.
- So it's more of an opt-in solution, than opt-out (if you set the rule, then disable for that Object/line only).

**Are you willing to submit a pull request to implement this change?**
- Not quite sure how to go about it myself, so not at this time no :cry:.

**Comment by nzakas** [comment]
As [noted in our README](https://github.com/eslint/eslint#stylistic-rule-updates), we are no longer adding options to rules that just enforce stylistic preferences. You can always copy the existing rule and modify it to create a custom rule to do what you want.
